main_plot.py

```diff
@@ -1,54 +1,109 @@
 #!/usr/bin/env python3
-
+if __name__ == "__main__":
+    from typing import (
+        Any,
+        Callable,
+        Dict,
+        List,
+        Optional,
+        Tuple,
+        Type,
+        Union,
+        Generator,
+    )
 import pandas as pd
 import numpy as np
+import logging
 from pathlib import Path
 from typing import TypeVar, List
+from predictions.Prediction import PredictionData
 
 from reports.report_loader import report_loader
+from reports.report_loader import json_report_loader
 from matplotlib import pyplot as plt
 import seaborn as sns
 import json
 import gzip
+from data.dataset import Dataset, Result
+from data.report import Report
+from methods.predict import Predict
 
 from plots.plot_results_in_heatmap import plot_results_in_heatmap_from_csv
 from plots.plot_results_in_box_plot import plot_results_in_boxplot_from_csv
+from plots.comparison_plot_multi import comparison_plot_multi
+
+from predictions.AR import ar
+from predictions.MA import ma
+from predictions.HoltWinters import holt_winters
+from predictions.ARIMA import arima
+from predictions.SARIMA import sarima
+from predictions.auto_arima import auto_arima
+from predictions.linear_regression import linear_regression
+from predictions.prophet import prophet
+from predictions.FCNN import fcnn
+from predictions.FCNN_embedding import fcnn_embedding
+from predictions.SES import ses
+from predictions.tsetlin_machine import tsetlin_machine, tsetlin_machine_single
+
+from data.list_of_tuples import list_of_tuples
+from data.airline_passengers import airline_passengers
+from data.sun_spots import sun_spots
+from data.data_from_csv import load_from_csv
+from data.load import Load
 
 from data.india_pollution import (
     india_pollution,
     get_list_of_city_names,
     get_list_of_coastal_indian_cities,
 )
 
+
 from data.stock_prices import (
     stock_prices,
     get_a_list_of_value_stock_tickers,
     get_a_list_of_growth_stock_tickers,
 )
 
+__testset_size = 0.2
+
 Data = TypeVar("Data", contravariant=True)
 Prediction = TypeVar("Prediction", covariant=True)
 ConfidenceInterval = TypeVar("ConfidenceInterval", covariant=True)
 Title = TypeVar("Title", covariant=True)
 Figure = TypeVar("Figure", covariant=True)
 
+
 from methods.plot import Plot
 
+__dataset_loaders: dict[str, Load[Dataset]] = {
+    "India city pollution": india_pollution,
```

Comment:
It seems rather wasteful to convert India to "Indian" and then back again, maybe check why you're doing this?